storage/bucket.go

```diff
@@ -864,6 +864,9 @@ type IPFilter struct {
 	// VPCNetworkSource: The list of VPC network
 	// (https://cloud.google.com/vpc/docs/vpc) sources of the bucket's IP filter.
 	VPCNetworkSource []VPCNetworkSource
+	// allow_all_service_agent_access: Whether or not to allow all P4SA access to the bucket.
+	// When set to true, IP filter config validation will not apply.
+	AllowAllServiceAgentAccess bool
```

Comment:
Let's discuss making this a pointer (forcing users to set this field explicitly) offline.